pkg/config/certificates.go

```diff
@@ -0,0 +1,151 @@
+package config
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+)
+
+type CertUser interface {
+	ClientCert() string
+	ClientKey() string
+	CACert() string
+	ClientCertPath() string
+	ClientKeyPath() string
+	CACertPath() string
+}
+
+func GetHTTPClient(certUser CertUser) (http.Client, error) {
+	timeout := 90 * time.Second
+
+	var cert []byte
+	if certUser.ClientCert() != "" {
+		cert = []byte(certUser.ClientCert())
+	} else if certUser.ClientCertPath() != "" {
+		file, err := os.ReadFile(certUser.ClientCertPath())
+		if err != nil {
+			return http.Client{}, err
+		}
+		cert = file
+	}
+
+	var key []byte
+	if certUser.ClientKey() != "" {
+		key = []byte(certUser.ClientKey())
+	} else if certUser.ClientKeyPath() != "" {
+		file, err := os.ReadFile(certUser.ClientKeyPath())
+		if err != nil {
+			return http.Client{}, err
+		}
+		key = file
+	}
+
+	var caCert []byte
+	if certUser.CACert() != "" {
+		caCert = []byte(certUser.CACert())
+	} else if certUser.CACertPath() != "" {
+		file, err := os.ReadFile(certUser.CACertPath())
+		if err != nil {
+			return http.Client{}, err
+		}
+		caCert = file
+	}
+
+	transport, err := GetTransport(cert, key, caCert, timeout)
+	if err != nil {
+		return http.Client{}, fmt.Errorf("error creating http transport: %w", err)
+	}
+
+	return http.Client{Transport: transport, Timeout: timeout}, nil
+}
+
+func GetTransport(certBytes, keyBytes, caCertBytes []byte, timeout time.Duration) (*http.Transport, error) {
+	transport := &http.Transport{ResponseHeaderTimeout: timeout}
+
+	if certBytes != nil && keyBytes != nil {
+		cert, err := tls.X509KeyPair(certBytes, keyBytes)
+		if err != nil {
+			return transport, fmt.Errorf("could not load keypair: %w", err)
+		}
+		tlsConfig := &tls.Config{
+			Certificates: []tls.Certificate{cert},
+			MinVersion:   tls.VersionTLS12,
+		}
+
+		if caCertBytes != nil {
+			pool, err := certPool(caCertBytes)
+			if err != nil {
+				return transport, err
+			}
+			tlsConfig.RootCAs = pool
+		}
+		transport.TLSClientConfig = tlsConfig
+	}
+	return transport, nil
+}
+
+func certPool(caCert []byte) (*x509.CertPool, error) {
+	pool := x509.NewCertPool()
+	ok := pool.AppendCertsFromPEM(caCert)
+	if !ok {
+		return nil, fmt.Errorf("could not parse candlepin ca cert")
```

Comment:
small suggested change to the error message since i think technically we're only dealing with candlepin ca certs here now, but will likely use these methods for more in the future?

```suggestion
		return nil, fmt.Errorf("could not parse ca cert")
```

Reply:
agreed!